src/main/java/com/powsybl/openloadflow/sa/AcSecurityAnalysis.java

```diff
@@ -120,6 +120,9 @@ private SecurityAnalysisResult runSimulations(LfNetwork network, List<Propagated
 
         // run pre-contingency simulation
         try (AcLoadFlowContext context = new AcLoadFlowContext(network, acParameters)) {
+            // save state for later restoration after base case and after each contingency
+            NetworkState networkState = NetworkState.save(network);
```

Comment:
This is not the right place to save the state. Here, you are loosing the benefits of the first resolution : the slack distribution, the switching of PV buses to PQ buses, the final tap position of transformers and the final section count of shunts. This has to be done after the `run`, for performance mainly.